src/traces/funnelarea/calc.js

```diff
@@ -0,0 +1,24 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var pieCalc = require('../pie/calc');
+
+function calc(gd, trace) {
+    return pieCalc.calc(gd, trace);
+}
+
+function crossTraceCalc(gd) {
+    pieCalc.crossTraceCalc(gd, { type: 'funnelarea' });
```

Comment:
I was skeptical at first, but looks like reusing `pie/calc.js` almost verbatim works fine for various `piecolorway` and `funnleareacolorway` combos. For example,

```js
var labels = ['Lead', 'Pipeline', 'Proposal', 'Negotiation', 'Closed (Won)'];
var values = [23, 15, 11, 6, 2.1];

var colorwayLight = ['#7fc97f', '#beaed4','#fdc086']
var colorwayDark = ['#1b9e77','#d95f02','#7570b3']

Plotly.newPlot(gd, [{
  type: 'pie',
  labels: labels,
  values: values,
  domain: {row: 0, column: 0}
}, {
  type: 'funnelarea',
  labels: labels,
  values: values,
  domain: {row: 0, column: 1}
}], {
  grid: {rows: 1, columns: 2},
  piecolorway: colorwayLight,
  funnelareacolorway: colorwayDark,
})
```

yields:

![image](https://user-images.githubusercontent.com/6675409/58048347-8d34f380-7b18-11e9-8161-33ecc63e4dfa.png)

generating the correct "extended" pie and funnelarea colors separately.

Nice job!

Reply:
@etpinard Thanks for testing this.
Included this mock in https://github.com/plotly/plotly.js/pull/3876/commits/de8b5176a605452b58f4144b02414cd722625ece.